### DISCUSSION TITLE
Question: getting input text from TTY::Editor.open(text: "")

### DISCUSSION THREAD

**Discussion by ognjenio** [piotrmurach/tty#80] [Q&A]
Hi, 

is there a way to get the text user entered when using `TTY::Editor.open(text: "")` without having to open a tempfile manually and then read it?

Sidenote: `examples/tempfile.rb` should be changed to be `TTY::Editor.open(text` instead of  `TTY::Editor.open(content`

**Comment by piotrmurach** [comment]
Hi Ognjen,

Currently the way you wish to do it, the answer is: no. Any time your provide `text:` option, you can also provide a filename as well which would help with retrieving info. 

The initial idea for `tty-editor` was to allow people to open and edit some configuration information. For entering multiline content via prompt, you could consider [tty-prompt](https://github.com/piotrmurach/tty-prompt) and the [multiline](https://github.com/piotrmurach/tty-prompt#23-multiline) prompt type.

Now, we could explore an improvement in `tty-editor` that would make it easier to access edited content. For example, we could yield the currently open file(s) inside a block:

```ruby
TTY::Editor.open(text: "") do |file|
  puts file.read  
end
```

Any thoughts?

**Comment by ognjenio** [comment]
Hi, 

I tried prompt and multiline and they don't really do the same thing. 

I got there by opening a temp file and then reading it once the editor is closed. It's fine just a bit messy. 

That implementation looks great.
